File: classify-server/src/main/java/io/ermdev/alice/entity/Role.java
package io.ermdev.alice.entity;

import mapfierj.NoRepeat;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@NoRepeat
@Table(name="tblrole")
@Entity
public class Role {

    @GeneratedValue(strategy = GenerationType.AUTO)
    @Id
    private Long id;

    private String name;

    @ManyToMany(mappedBy = "roles")
    private List<User> users = new ArrayList<>();

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<User> getUsers() {
        return users;
    }

    public void setUsers(List<User> users) {
        this.users = users;
    }

    @Override
    public String toString() {
        return "Role{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", users=" + users +
                '}';
    }
}

File: src/main/kotlin/io/ermdev/classify/data/entity/Role.kt
package io.ermdev.classify.data.entity

import javax.persistence.*

@Entity
@Table(name = "tbl_role")
class Role(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long = 0,

        var name: String = "",

        @ManyToMany(mappedBy = "roles", cascade = [CascadeType.REMOVE])
        var users: MutableSet<User> = HashSet())